### PR TITLE
L2 verifier support

### PR DESCRIPTION
**Pull request by StanislavBreadless**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l2-contracts/hardhat.config.ts [hunk 1]
```diff
@@ -12,14 +12,14 @@ if (!process.env.CHAIN_ETH_NETWORK) {
 
 export default {
   zksolc: {
-    version: "1.3.18",
+    version: "1.5.1",
```
Comment (vladbochok) [inline l2-contracts/hardhat.config.ts:15]: Why not `1.5.3`?

File: l2-contracts/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
   },
   "scripts": {
     "build": "hardhat compile",
-    "test:foundry": "forge test --zksync",
+    "test:foundry": "forge test --zksync --gas-limit 2000000000",
```
Comment (vladbochok) [inline l2-contracts/package.json:36]: Seems like you add config var for gas limit in the foundry.toml and here

File: tools/src/main.rs [hunk 3]
```diff
@@ -334,3 +344,37 @@ fn generate_residue_g2_elements(vk: &HashMap<String, Value>) -> String {
 
     residue_g2_elements
 }
+
+
+fn get_modexp_function(l2_mode: bool) -> String {
```
Comment (vladbochok) [inline tools/src/main.rs:349]: Very nit with low priority: I would want to make foundry fuzz test to catch that these two functions behave exactly the same way.

**Review by vladbochok — APPROVED** [review]

File: tools/src/main.rs (hunk 3 above)
Comment (StanislavBreadless) [reply tools/src/main.rs:349]: Okay, I'll create a task for that

File: l2-contracts/package.json (hunk 2 above)
Comment (StanislavBreadless) [reply l2-contracts/package.json:36]: I removed from the config (this value has to be passed here)

**Pull request merged**